apps/vanilla/src/pages/exercise2/scripts/domain/activePlayerLoop.ts

```diff
@@ -0,0 +1,62 @@
+import { Player } from './player';
+
+/**
+ * Collection of players that return only player who not pass.
+ */
+export class ActivePlayerLoop implements IterableIterator<Player | null> {
+
+	private readonly players: Player[];
+
+	private currentIdex = -1;
+
+	public constructor(players: Player[]) {
+		this.players = players;
+	}
+
+	private get passedPlayerIndexes(): number[] {
+		return this.players
+			.filter(player => player.passStatus)
+			.map(player => this.players.indexOf(player));
+	}
+
+	/**
+	 * Provides there is any player who not pass.
+	 */
+	public get activePlayersEnded(): boolean {
+		return this.passedPlayerIndexes.length === this.players.length;
+	}
+
+	/**
+	 * Get next active player.
+	 */
+	public next(): IteratorResult<Player | null> {
+		if (this.activePlayersEnded) {
+			return {
+				done: false,
+				value: null,
+			};
+		}
+
+		this.currentIdex++;
+
+		if (this.currentIdex === this.players.length) {
```

Comment:
Please do not use [contractions](https://wiki.saritasa.rocks/frontend/languages/ts-js/naming#avoid-contractions)
```suggestion
		if (this.currentIndex === this.players.length) {
```